packages/table/src/common/loading.tsx

```diff
@@ -0,0 +1,24 @@
+/**
+ * Copyright 2016 Palantir Technologies, Inc. All rights reserved.
+ * Licensed under the BSD-3 License as modified (the “License”); you may obtain a copy
+ * of the license at https://github.com/palantir/blueprint/blob/master/LICENSE
+ * and https://github.com/palantir/blueprint/blob/master/PATENTS
+ */
+
+import * as React from "react";
+
+export interface ILoadable {
+    /**
+     * Show an animated loading animation.
+     * @default false
+     */
+    isLoading?: boolean;
+}
+
+export function renderLoadingSkeleton(baseClassName: string) {
+    const width = Math.floor(Math.random() * 4) * 5;
+    const skeletonClassName = `${baseClassName}-skeleton`;
+    return (
+        <div className={`${skeletonClassName} ${skeletonClassName}-${width}`} />
+    );
+}
```

Comment:
This seems like this should become a React component, and it should live in @blueprintjs/core, or at the very least be made into an independent component in @blueprintjs/table and then move to core later.

I think what I want is something like:

```
<LoadingSkeleton isLoading={true}>
    <div>loaded contents or whatever</div>
</LoadingSkeleton>
```

And then is `isLoading` is true, we could clone the child, but substitute our own children. We would have to assume that there is always exactly one children, but we use this pattern already for e.g. `DragSelectable`.